### PR TITLE
Added Dockerfile to contrib with README

### PR DESCRIPTION
**Pull request by benzcash**
Please ensure this checklist is followed for any pull requests for this repo. This checklist must be checked by both the PR creator and by anyone who reviews the PR.
* [ ] Relevant documentation for this PR has to be completed and reviewed by @mdr0id before the PR can be merged
* [ ] A test plan for the PR must be documented in the PR notes and included in the test plan for the next regular release

As a note, all buildbot tests need to be passing and all appropriate code reviews need to be done before this PR can be merged

### CODE REVIEW

File: contrib/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+```
+mkdir {./zcash-params-dir,./zcash-data-dir}
+sudo chown -R 2001.2001 {./zcash-params-dir,./zcash-data-dir}
+docker run -d --name my_zcashd \
+  -v $(pwd)/zcash-data-dir:/srv/zcashd/.zcash \
+  -v $(pwd)/zcash-params-dir/srv/zcashd/.zcash-params \
+  electriccoinco/zcashd
+```  
+
+Follow the logs to see it's activity
```
Comment (daira) [inline contrib/docker/README.md:87]:
```suggestion
Follow the logs to see its activity.
```

File: contrib/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    fi; \
+    zcashd --version
+
+RUN useradd --home-dir /srv/$ZCASHD_USER \
+            --shell /bin/bash \
+            --create-home \
+            --uid $ZCASHD_UID\
+            $ZCASHD_USER
+USER $ZCASHD_USER
+WORKDIR /srv/zcashd
```
Comment (charlieok) [inline contrib/docker/Dockerfile:32]: The substitution `srv/$ZCASHD_USER` is used in the other places this directory is referenced. Maybe do that here too?

File: contrib/docker/README.md (hunk 1 above)
Comment (benzcash) [reply contrib/docker/README.md]: Updated!

File: contrib/docker/Dockerfile (hunk 2 above)
Comment (benzcash) [reply contrib/docker/Dockerfile]: Yeah, it should match the valued used for `--home-dir /srv/$ZCASHD_USER `

Updated!

File: contrib/docker/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# zcash Dockerfile
+
+This is an example Dockerfile to run the zcashd daemon in a containerized Debian base OS. The image pulls the apt package built by Electric Coin Company.
+
+The output of building this image should be accessible at https://hub.docker.com/repository/docker/electriccoinco/zcashd
```
Comment (charlieok) [inline contrib/docker/README.md:5]: This url isn't accessible without logging into docker hub. Would a url like https://hub.docker.com/r/electriccoinco/zcashd be a good substitute here?

File: contrib/docker/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+## Defaults
+
+The image will run as a non-root user, `zcashd` with uid `2001`. When mapping volumes from the host into the container, these permissions must be used, or rebuilt the image with your custom values.
```
Comment (charlieok) [inline contrib/docker/README.md:9]: “rebuilt the image” -> “rebuild the image” or “the image rebuilt”

File: contrib/docker/README.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+## Run time configuration
+
+The goal is to follow the default zcashd startup behavior as closely as possible.
+
+At startup, the image will execute the [./entrpoint.sh](./entrypoint) script. This script use environmental variables to configure the command line parameters, do a little house cleaning, and start zcashd.
```
Comment (charlieok) [inline contrib/docker/README.md:15]: “entrpoint.sh” -> “entrypoint.sh”

Comment (charlieok) [inline contrib/docker/README.md:15]: “This script use” -> “This script uses”

File: contrib/docker/README.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+or <https://www.opensource.org/licenses/mit-license.php>.
+
+This product includes software developed by the OpenSSL Project for use in the
+OpenSSL Toolkit <https://www.openssl.org/> and cryptographic software written
+by Eric Young.
+```
+
+### Persist data to the host
+
+For this example, we'll create a a place for zcashd to store the blockchain data, create a new container that uses that location, and run it in the background.
```
Comment (charlieok) [inline contrib/docker/README.md:76]: “create a a place” -> “create a place”

File: contrib/docker/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+RUN mkdir -p /srv/$ZCASHD_USER/.zcash/ \
+    && touch /srv/$ZCASHD_USER/.zcash/zcash.conf
```
Comment (charlieok) [inline contrib/docker/Dockerfile:34]: The `entrypoint.sh` script runs `touch .zcash/zcash.conf` near the end. Do we want to do that both here and there?

Comment (benzcash) [reply contrib/docker/Dockerfile:34]: This is a "convenience" in case you skip the entrypoint script.

File: contrib/docker/README.md (hunk 3 above)
Comment (benzcash) [reply contrib/docker/README.md]: Doh, updated!

File: contrib/docker/README.md (hunk 5 above)
Comment (benzcash) [reply contrib/docker/README.md]: Thanks, updated.

File: contrib/docker/README.md (hunk 6 above)
Comment (benzcash) [reply contrib/docker/README.md]: Thanks, updated.

File: contrib/docker/README.md (hunk 5 above)
Comment (benzcash) [reply contrib/docker/README.md]: Thanks, updated.

File: contrib/docker/README.md (hunk 4 above)
Comment (benzcash) [reply contrib/docker/README.md]: Thanks, updated.

**Comment by mdr0id** [comment]
ACK

**Comment by ebfull** [comment]
ACK since this isn't touching any code, but didn't look into it in super detail.

@zkbot r+

**Pull request merged**